### PR TITLE
New Design Added

### PR DESCRIPTION
**Pull request by lalitmee**

### CODE REVIEW

File: download.js [hunk 1]
```diff
@@ -270,16 +275,58 @@ function injectButtons () {
...
+      if (videos[j] && videos[j].poster === dlink) {
+        if (videos[j] && videos[j].src) {
+          dlink = videos[j].src
+        }
+        j += 1
+      }
+
+      dlbutton.innerHTML = `
+                <a download href=${escapeHTML(dlink)}>
+                <button class="download-button instanshu-unite instanshu-sm instanshu-success">
```
Comment (aashutoshrathi) [inline download.js]: Give single class to buttons which is `class="dCJp8 afkep _0mzm-"`

File: insta-down-css.css [hunk 2]
```diff
@@ -438,3 +438,11 @@
   align-items: center;
   justify-content: space-around;
 }
+
+.download-button {
```
Comment (aashutoshrathi) [inline insta-down-css.css]: No need of this change in CSS, I think.

**Comment by aashutoshrathi** [comment]
For Videos use this:
```
<img class="download-icon" src="https://img.icons8.com/ios/50/000000/sending-video-frames.png" width="29">
```

File: download.js [hunk 3]
```diff
@@ -77,12 +77,18 @@ function injectButtons () {
...
+
   // Grab images appearing on the page
   const images = document.querySelectorAll('img')
 
   for (i = 0; i < images.length; i += 1) {
     // making the image link a downloading link
-    if (!images[i].src.endsWith('&dl=1')) {
+    if (
+      !images[i].src.endsWith('&dl=1') &&
+      images[i].className !== 'download-icon'
```
Comment (aashutoshrathi) [inline download.js]: Add the same condition for video too.

**Comment by lalitmee** [comment]
The same thing is happening now, the videos order is somehow stable for the first 2 or 3 loads.

**Comment by aashutoshrathi** [comment]
I don't know why but few buttons still show download the download image :laughing: 
And also #50 is a pain too.

**Comment by aashutoshrathi** [comment]
I'm thinking of a major change since we can't differentiate between Profile Pictures and one I mentioned in #50.
I think we should remove the concept of profile image, for now, i.e. when querying the document for 'img' we should not take images of width=150px, and hence the Main page will have no bugs at all.

**Comment by lalitmee** [comment]
Actually, I think that if someone is interested in Profile pic then he/she can go to that user's profile and download a hell lot of pictures from there. 

I think its a good idea. :+1:

**Comment by aashutoshrathi** [comment]
Yes, @lalitmee should we do it in same PR.

**Comment by lalitmee** [comment]
Yeah, we can.

**Comment by lalitmee** [comment]
I am trying to stop these two buttons, but I am not getting. I tried to put the condition with the class name of the newly created element but still, it is adding two.

File: download.js [hunk 4]
```diff
@@ -270,16 +258,65 @@ function injectButtons () {
...
+
+      const btns = sections[i].querySelectorAll('span')
+
+      if (btns.length === 6) {
+        if (sections[i].querySelectorAll('span').length < 7) {
+          sections[i].insertBefore(downloadButton, btns[4])
+        }
+      } else {
+        if (sections[i].querySelectorAll('span').length < 9) {
+          sections[i].insertBefore(downloadButton, btns[6])
```
Comment (aashutoshrathi) [inline download.js]: This is getting triggered for second button. :sweat_smile:

Comment (lalitmee) [reply download.js]: But now, I am checking for the length of the `span` and if there is not `share` button then it means only 6 spans are there. Then it should not come in the else condition.

**Comment by lalitmee** [comment]
Hey @aashutoshrathi, do you want me to do anything else in this PR. I haven't removed profile images because currently, I am not facing any error in the console. If you want me to remove those images and **DP** button, I can remove.

**Comment by aashutoshrathi** [comment]
Things working fine now?
One more check and we are good to go.

**Comment by aashutoshrathi** [comment]
/remind me to review and merge this PR in 1 day

**Comment by lalitmee** [comment]
Hey @aashutoshrathi, I think we can merge this.

**Pull request merged**